master/internal/rm/multirm/multirm_intg_test.go

```diff
@@ -0,0 +1,587 @@
+
+// TestMain sets up the DB for tests.
+func TestMain(m *testing.M) {
+	tmp, err := db.ResolveTestPostgres()
+	pgDB = tmp
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = db.MigrateTestPostgres(pgDB, "file://../../../static/migrations", "up")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	err = etc.SetRootPath("../../../static/srv")
+	if err != nil {
+		log.Panicln(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestGetAllocationSummaries(t *testing.T) {
+	cases := []struct {
+		name       string
+		allocNames []string
+		managers   int
+	}{
+		{"simple", []string{uuid.NewString(), uuid.NewString()}, 1},
+		{"multirm", []string{uuid.NewString(), uuid.NewString()}, 3},
+	}
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			rms := map[string]rm.ResourceManager{}
+			for i := 1; i <= tt.managers; i++ {
+				ret := map[model.AllocationID]sproto.AllocationSummary{}
+				for _, alloc := range tt.allocNames {
+					a := alloc + fmt.Sprint(i)
+					ret[*model.NewAllocationID(&a)] = sproto.AllocationSummary{}
+				}
+				require.Equal(t, len(ret), len(tt.allocNames))
+
+				mockRM := mocks.ResourceManager{}
+				mockRM.On("GetAllocationSummaries").Return(ret, nil)
+
+				rms[uuid.NewString()] = &mockRM
+			}
+
+			m := &MultiRMRouter{rms: rms}
+
+			allocs, err := m.GetAllocationSummaries()
+			require.NoError(t, err)
+			require.Equal(t, tt.managers*len(tt.allocNames), len(allocs))
+			require.NotNil(t, allocs)
+
+			bogus := "bogus"
+			require.Empty(t, allocs[*model.NewAllocationID(&bogus)])
+
+			for _, name := range tt.allocNames {
+				n := fmt.Sprintf(name + "0")
+				tmpName := name
+
+				require.NotNil(t, allocs[*model.NewAllocationID(&n)])
+				require.Empty(t, allocs[*model.NewAllocationID(&tmpName)])
+			}
+		})
+	}
+}
+
+func TestAllocate(t *testing.T) {
+	mockRM := mocks.ResourceManager{}
+	manager := uuid.NewString()
+	m := &MultiRMRouter{rms: map[string]rm.ResourceManager{manager: &mockRM}}
+
+	allocReq := sproto.AllocateRequest{ResourceManager: manager}
+	mockRM.On("Allocate", manager, allocReq).Return(&sproto.ResourcesSubscription{}, nil)
+
+	res, err := m.Allocate(manager, allocReq)
+	require.NoError(t, err)
+	require.Equal(t, res, &sproto.ResourcesSubscription{})
+
+	res, err = m.Allocate("bogus", sproto.AllocateRequest{})
+	require.Equal(t, err, ErrRPNotDefined("bogus", ""))
+	require.Nil(t, res)
+}
+
+func TestValidateResources(t *testing.T) {
+	mockRM := mocks.ResourceManager{}
+	manager := uuid.NewString()
+	m := &MultiRMRouter{rms: map[string]rm.ResourceManager{manager: &mockRM}}
+
+	req := sproto.ValidateResourcesRequest{
+		ResourcePool: "",
+		Slots:        0,
+		IsSingleNode: true,
+	}
+
+	mockRM.On("ValidateResources", manager, req).Return(nil, nil)
+
+	_, err := m.ValidateResources(manager, req)
+	require.NoError(t, err)
+
+	_, err = m.ValidateResources("bogus", sproto.ValidateResourcesRequest{})
+	require.Equal(t, err, ErrRPNotDefined("bogus", ""))
+}
+
+func TestDeleteJob(t *testing.T) {
```

Comment:
What happens if it attempts to delete a job that doesn't exist?

Reply:
Well, good news, DeleteJob isn't implemented for Agent or K8s resource managers, so this is kind of a moot point for this implementation of MultiRM -- that's why I'm not even taking in a rmName field to look up in the rms map.
This will be something @stoksc will address when we expand multiRM to dispatcherrm (slurm)